### ISSUE TITLE
esbuild-jest do not works with reflection system.

### ISSUE THREAD

**Issue by gustawdaniel** [MichalLytek/typegraphql-prisma#277]
I typed a simple test in jest.

```
import {getApolloServer} from "../src/server";
import pJson from '../package.json';

it('i can read version from api', async () => {
    const server = await getApolloServer();
    const result = await server.executeOperation({
        query: `{
  version {
    name
    version
  }
}`,
    });

    expect(result.errors).toBeUndefined();
    expect(result.data?.version).toEqual({name: pJson.name, version: pJson.version});
})
```

It works with `ts-jest`, (jest.config.js file)

```
module.exports = {
  "roots": [
    "<rootDir>"
  ],
  "testMatch": [
    "**/__tests__/**/*.+(ts|tsx)",
    "**/?(*.)+(spec|test).+(ts|tsx)"
  ],
  "transform": {
    "^.+\\.(ts|tsx)$": "ts-jest"
  },
  "setupFilesAfterEnv": ["jest-extended"],
  "testEnvironment": "node"
}
```

but if I replace the "transform" option with about 25 times more efficient "esbuild-jest". I mean `jest.config.js` with

```
module.exports = {
  "roots": [
    "<rootDir>"
  ],
  "testMatch": [
    "**/__tests__/**/*.+(ts|tsx)",
    "**/?(*.)+(spec|test).+(ts|tsx)"
  ],
  "transform": {
    "^.+\\.(ts|tsx)$": "esbuild-jest"
  },
  "setupFilesAfterEnv": ["jest-extended"],
  "testEnvironment": "node"
}
```

than I see errors:

```
Unable to infer GraphQL type from TypeScript reflection system. You need to provide explicit type for parameter #2 of 'findUniqueEvent_values' of 'Event_valuesCrudResolver' class.
```

<img width="1373" alt="Screenshot 2022-04-15 at 20 06 15" src="https://user-images.githubusercontent.com/16663028/163593747-fd2f6536-5fac-410a-8ced-0037e413d097.png">

For manually typed resolvers it requires adding "type" function to all fields and arguments.

I mean - this works with ts-jest

```
    @Field({ nullable: true })
    phone?: string;
```

but esbuild-jest requires this form

```
    @Field(()=>String, { nullable: true })
    phone?: string;
```

If we would be able to solve this problem and add `esbuild-jest` support it can speed up testing many times.

**Comment by MichalLytek** [comment]
@gustawdaniel You can setup ts-jest to not check the typings but only transpile the codebase. It's then a few times faster 😉 

I may add an option to emit full, redundant type definition but it might increase generated code size. So I would make it optional and enabled via generator options.

**Comment by tsteenkamp** [comment]
> I may add an option to emit full, redundant type definition but it might increase generated code size. So I would make it optional and enabled via generator options.

That would be really awesome!
At the moment the above error also stops us from using the awesome [vitest](https://vitest.dev/) framework (or anything else based on esbuild, I guess).

**Comment by MichalLytek** [comment]
Closing as fixed by 36113bee 🔒 Please use the new `emitRedundantTypesInfo` generator option.
